application/libraries/Aauth.php

```diff
@@ -121,42 +121,40 @@ public function __construct() {
 		$this->errors = $this->CI->session->flashdata('errors') ?: array();
 		$this->infos = $this->CI->session->flashdata('infos') ?: array();
 
-		// Pre-Cache IDs
-		$this->precache_ids();
-
-	}
-
-	/**
-	 * pre_cache_ids() caches all permission and group IDs for later use.
-	 */
-	public function precache_ids() {
-
 		// Initialize Variables
 
 		$this->cache_perm_id		= array();
 		$this->cache_group_id		= array();
+		
+		// Pre-Cache IDs
+		$this->precache_perms();
+		$this->precache_groups();
 
-		// Permissions
-
+	}
+	
+	/**
+	 * precache_perms() caches all permission IDs for later use.
+	 */
+	private function precache_perms {
 		$query	= $this->aauth_db->get($this->config_vars['perms']);
 
 		foreach ($query->result() as $row) {
 			$key				= str_replace(' ', '', trim(strtolower($row->name)));
 			$this->cache_perm_id[$key]	= $row->id;
-		}
-
-		// Groups
-
+	}
+	
+	/**
+	 * precache_groups() caches all group IDs for later use.
+	 */
+	private function precache_groups {
```

Comment:
The function name line should be as follows:
private function precache_groups() {

Observer the missing '()' after the function name

Reply:
This will address issue #234